test/e2e/local_ratelimit_test.go

```diff
@@ -0,0 +1,344 @@
+package e2e_test
+
+import (
+	"fmt"
+	"net/http"
+
+	v1 "github.com/solo-io/gloo/projects/gateway/pkg/api/v1"
+	gatewaydefaults "github.com/solo-io/gloo/projects/gateway/pkg/defaults"
+	gloov1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	gloo_matchers "github.com/solo-io/gloo/projects/gloo/pkg/api/v1/core/matchers"
+	"github.com/solo-io/gloo/projects/gloo/pkg/api/v1/enterprise/options/ratelimit"
+	"github.com/solo-io/gloo/projects/gloo/pkg/api/v1/options/local_ratelimit"
+	local_ratelimit_plugin "github.com/solo-io/gloo/projects/gloo/pkg/plugins/local_ratelimit"
+	"github.com/solo-io/gloo/test/e2e"
+	"github.com/solo-io/gloo/test/gomega/matchers"
+	"github.com/solo-io/gloo/test/testutils"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Local Rate Limit", func() {
+
+	const (
+		defaultLimit = 3
+		vsLimit      = 2
+		routeLimit   = 1
+	)
+
+	var (
+		testContext *e2e.TestContext
+
+		httpClient                            *http.Client
+		requestBuilder                        *testutils.HttpRequestBuilder
+		expectSuccess                         func()
+		expectRateLimitedWithXRateLimitHeader func(int)
+	)
+
+	BeforeEach(func() {
+		testContext = testContextFactory.NewTestContext()
+		testContext.BeforeEach()
+
+		httpClient = testutils.DefaultHttpClient
+		requestBuilder = testContext.GetHttpRequestBuilder()
+
+		expectSuccess = func() {
```

Comment:
can we rename this?  when i read this function i expect that we are succeeeding on whats tested but what we are really doing here is expected an ok non-ratelimited response

Perhaps 
```suggestion
		expectNotLimited = func() {
```

Reply:
Makes sense, fixed